libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -65,15 +65,6 @@ class MapboxDirectionsSession(
         adjustedRouteOptions: RouteOptions,
         routesRequestCallback: RoutesRequestCallback
     ) {
-        // TODO What happens if NavigationSession.State transitions from Active Guidance
-        //  to Free Drive and Faster Route is enabled? We're not cleaning up RouteOptions when
-        //  cleaning up routes here and this call can be done because the only thing that we're
-        //  checking is the route options and the route progress but not if there's a route or not
-        //  See AdjustedRouteOptionsProvider#getRouteOptions
-        if (routes.isEmpty()) {
-            routesRequestCallback.onRoutesRequestCanceled(adjustedRouteOptions)
```

Comment:
Choosing to remove a cancel callback when expected behavior occurs

Reply:
i don't see a reason a customer would ever want to know that we didn't request faster routes, because there isn't a current route

Reply:
also a cancel callback indicates that we made a request to the api (which impacts billing), and ignored the response for some reason. that kind of callback usually falls into a failure callback. 

Should we delete the onRoutesRequestCanceled?